processor/batchprocessor/batch_processor.go

```diff
@@ -404,6 +406,7 @@ func newMetricsBatchProcessor(set processor.Settings, next consumer.Metrics, cfg
 
 // ConsumeMetrics implements processor.Metrics
 func (m *metricsBatchProcessor) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	pref.RefMetrics(md)
```

Comment:
Do you think an interface like this make it less error prone?
```
releaseCallback := pref.ClaimMetrics(md)
defer releaseCallback()
```

Reply:
I don't think other components need to do this (I checked, only less than 5 need to do it), so this is more efficient because we need to ref in one place and unref in another (look at exporterhelper) so then I need to pass around queues/etc and extra obeject to defer on it.

Reply:
Yeah, makes sense.